src/test/java/picard/sam/AddOATagTest.java

```diff
@@ -0,0 +1,142 @@
+package picard.sam;
+
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.SAMRecordIterator;
+import htsjdk.samtools.SamReaderFactory;
+import htsjdk.samtools.util.Interval;
+import htsjdk.samtools.util.OverlapDetector;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class AddOATagTest {
+    @DataProvider(name="testOAData")
+    Object[][] testOAData() {
+        return new Object[][]{
+                new Object[]{new File("testdata/picard/sam/aligned_withoutOA.sam"), new File("testdata/picard/sam/aligned_withOA.sam")},
+        };
+    }
+
+    @DataProvider(name="testIntervalListData")
+    Object[][] testIntervalListData() {
+        return new Object[][]{
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), new File("testdata/picard/analysis/directed/CollectHsMetrics/chrM.interval_list")},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), new File("testdata/picard/sam/contiguous.interval_list")}
+        };
+    }
+
+    @DataProvider(name="testOverwriteData")
+    Object[][] overwriteTestData() {
+        return new Object[][]{
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null, true},
+                new Object[]{new File("testdata/picard/sam/aligned.sam"), null, false}
```

Comment:
Because the default is overwrite, I guess that the test cases here should be for not overwriting, with and without interval list and with/without original OA tag.